scripts/deployment/deploy_env/chain_env.sh

```diff
@@ -12,92 +12,13 @@ function init_chain_env() {
-		;;
-
-	juno)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/mainnets/juno.env)
-		;;
-
-	juno-testnet)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/testnets/juno.env)
-		;;
-
-	terra)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/mainnets/terra.env)
-		;;
-
-	terra-testnet)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/testnets/terra.env)
-		;;
-
-	archway-testnet)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/testnets/archway.env)
-		;;
-
-	chihuahua)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/mainnets/chihuahua.env)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/base_chihuahua.env)
-		;;
-
-	injective)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/mainnets/injective.env)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/base_injective.env)
-		;;
-
-	injective-testnet)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/testnets/injective.env)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/base_injective.env)
-		;;
-
-	comdex)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/mainnets/comdex.env)
-		;;
-
-	comdex-testnet)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/testnets/comdex.env)
-		;;
-
-	sei-testnet)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/testnets/sei.env)
-		;;
-
-	stargaze-testnet)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/testnets/stargaze.env)
-		;;
-
-	migaloo)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/mainnets/migaloo.env)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/base_migaloo.env)
-		;;
-
-	migaloo-testnet)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/testnets/migaloo.env)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/base_migaloo.env)
-		;;
-
-	osmosis)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/mainnets/osmosis.env)
-		;;
-
-	osmosis-testnet)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/testnets/osmosis.env)
-		;;
-
-	orai)
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/mainnets/orai.env)
-		;;
-
-	*)
-		echo "Network $chain not defined"
-		exit 1
-		;;
-	esac
-
-	if [[ $chain != "chihuahua" && $chain != "injective" && $chain != "injective-testnet" && $chain != "migaloo" && $chain != "migaloo-testnet" ]]; then
-		source <(cat "$project_root_path"/scripts/deployment/deploy_env/base.env)
+	if [[ "$(echo ${chain##*-})" = "testnet" ]] || [[ "$chain" = "local" ]]; then
+		chain="${chain%-testnet}"
+		source <(cat "$project_root_path"/scripts/deployment/deploy_env/testnets/"$chain".env)
+	else
+		source <(cat "$project_root_path"/scripts/deployment/deploy_env/mainnets/"$chain".env)
 	fi
+
```

Comment:
yes that should work, the chain_env script remained "unchanged" for the most, the only logical difference is that when it can't find the chain it does return 1 instead of exit 1. exit 1 was killing the terminal which you might not want.